src/main/java/com/nscharrenberg/um/multiagentsurveillance/headless/models/TileArea.java

```diff
@@ -26,63 +23,103 @@ public boolean within(int x, int y) {
     }
 
     @Override
-    public List<Tile> subset(int x1, int y1, int x2, int y2) {
-        List<Tile> subset = region.stream().filter(tile -> tile.getX() >= x1 && tile.getX() <= x2 && tile.getY() >= y1 && tile.getY() <= y2).toList();
+    public HashMap<Integer, HashMap<Integer, Tile>> subset(int x1, int y1, int x2, int y2) {
+        HashMap<Integer, HashMap<Integer, Tile>> generatedSubset = new HashMap<>();
+
+        for (Map.Entry<Integer, HashMap<Integer, Tile>> rowEntry : region.entrySet()) {
+            if (rowEntry.getKey() >= x1 && rowEntry.getKey() <= x2) {
+                for (Map.Entry<Integer, Tile> colEntry : rowEntry.getValue().entrySet()) {
+                    if (colEntry.getKey() >= y1 && colEntry.getKey() <= y2) {
+
+                        if (!generatedSubset.containsKey(rowEntry.getKey())) {
+                            generatedSubset.put(rowEntry.getKey(), new HashMap<>());
```

Comment:
Inserting unspecified hashmap here on purpose or do we not care which type it is here anyway?